### PR TITLE
correct licensing and incorporation of FastMath

### PR DESCRIPTION
**Pull request by talevy**
this resolves incorrectly licensed code in #49009.

ESSloppyMath is removed in favor of preserving as much of the original
FastMath as possible. Since no additional methods are introduced in
ESSloppyMath, this abstraction is removed.

thanks @jasontedor for assisting to resolve this.

### CODE REVIEW

**Review by jasontedor — COMMENTED** [review]
This is good. Thanks. I think I like a level of indirection on top of `FastMath` (and make it package private) though so that we aren't tempted to add code to it that doesn't fall under the license at the top of the file. Maybe I'm overthinking it, and I can be convinced otherwise.

**Comment by talevy** [comment]
I totally get the reasoning, but it feels like overkill given where that there is no other code that would be defined in ESSloppyMath, other than these thin wrappers. Would an alternative to introducing a wrapper be moving FastMath to its original package `org.math.plot.utils`?

**Comment by jasontedor** [comment]
I'm fine without the level of indirection if we add a top-level comment to the class that additions/modifications to it should only be those that come from the original source.

**Comment by talevy** [comment]
after working on some other code, I see value in introducing more helper methods like an `ESSloppyMath.sin` that is backed by `SloppyMath.cos`. So I will actually change my mind and revert back to your original recommendation, which will help build a home for these helpers

**Comment by jasontedor** [comment]
That’s great, thanks!

File: libs/core/src/main/java/org/elasticsearch/common/util/FastMath.java
Comment (jasontedor) [inline libs/core/src/main/java/org/elasticsearch/common/util/FastMath.java]: This class can be made package private now.

**Review by jasontedor — APPROVED** [review]
I left one comment that requires attention before merging, otherwise LGTM.

File: libs/core/src/main/java/org/elasticsearch/common/util/FastMath.java
Comment (talevy) [reply libs/core/src/main/java/org/elasticsearch/common/util/FastMath.java]: 👍

**Comment by talevy** [comment]
@elasticmachine update branch

**Comment by ywelsch** [comment]
The backport PR seems to have been merged. I'm therefore removing the backport pending label here. Please shout if this is incorrect

**Pull request merged**
